cmd/puma-dev/main_test.go

```diff
@@ -293,4 +305,24 @@ func runPlatformAgnosticTestScenarios(t *testing.T) {
 
 		assert.Equal(t, "rack wuz here", getURLWithHost(t, reqURL, statusHost))
 	})
+
+	t.Run("request-headers-dump contains X-Forwarded-* headers with http", func(t *testing.T) {
+		reqURL := fmt.Sprintf("http://localhost:%d/", *fHTTPPort)
+		statusHost := "request-headers-dump"
+
+		dumpedHeaders := parseDumpedHTTPHeadersFromBody(getURLWithHost(t, reqURL, statusHost))
+
+		assert.Equal(t, "127.0.0.1", dumpedHeaders["HTTP_X_FORWARDED_FOR"])
+		assert.Equal(t, "http", dumpedHeaders["HTTP_X_FORWARDED_PROTO"])
+	})
+
+	t.Run("request-headers-dump contains X-Forwarded-* headers with https", func(t *testing.T) {
+		reqURL := fmt.Sprintf("https://localhost:%d/", *fTLSPort)
```

Comment:
requesting HTTPS won't work here, since the test runner won't trust the cert.

but, since we don't care about verifying certs, we can just ignore it in [`getURLWithHost`](https://github.com/puma/puma-dev/pull/299/files#diff-0744ac226a19e0cfaccca8178f35cc1b117b23ab799d7fb4cdde458195bcdf9bL143).

```golang
// we don't care about checking certs in test
// the generated cert will not be trusted by the test runner
insecureTransport := &http.Transport{
        TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
}
insecureClient := &http.Client{Transport: insecureTransport}

resp, err := insecureClient.Do(req)
```